source/slang/lower-to-ir.cpp

```diff
@@ -1847,6 +1857,63 @@ struct StmtLoweringVisitor : StmtVisitor<StmtLoweringVisitor>
         insertBlock(breakLabel);
     }
 
+    void visitDoWhileStmt(DoWhileStmt* stmt)
+    {
+        // Generating IR for `do {...} while` statement is similar to a
+        // `while` statement, just with the test in a different place
+
+        auto builder = getBuilder();
+
+        // We will create blocks for the various places
+        // we need to jump to inside the control flow,
+        // including the blocks that will be referenced
+        // by `continue` or `break` statements.
+        auto loopHead = createBlock();
+        auto breakLabel = createBlock();
+
+        // A `continue` inside a `do { ... } while` loop always
+        // jumps to the head of the loop.
+        auto continueLabel = loopHead;
```

Comment:
Shouldn't a `continue` statement jump to the begining of condition evaluation?
If continue jumps to loop head, then any `continue` statement in a do-while loop will cause infinite loop, right?